### ISSUE TITLE
Failed to detect `markuplint.config.ts` automatically

### ISSUE THREAD

**Issue by munierujp** [markuplint/markuplint#948]
- Markuplint version: `3.9.0`
- Parser lang: HTML
- Node.js version: `18.14.2`
- OS: macOS

## Describe the bug

`markuplint.config.ts` is not detected automatically.

## Steps To Reproduce

I have `markuplint.config.ts` in my project root.

When I run the following script, the configuration file was not used.

```sh
npx markuplint "**/*.{html,vue}"
```

When run with the `--config` option, a configuration file was used.

```sh
npx markuplint --config markuplint.config.ts "**/*.{html,vue}"
```

## The current behavior

`markuplint.config.ts` will not be detected automatically.

## The expected behavior

`markuplint.config.ts` will be detected automatically.

**Comment by munierujp** [comment]
This issue may be related to #919

**Comment by risu729** [comment]
I was also looking into this issue.

It seems ESM is not supported at all. ([bug: fails to load config in ESM project #60](https://github.com/Codex-/cosmiconfig-typescript-loader/issues/60))

Also, I tried to apply the CJS config file using the `--config` option, but it didn't work.

## Steps to reproduce

`npx markuplint --config markuplint.config.ts --verbose test.html`

```ts:markuplint.config.ts
import type { Config } from "@markuplint/ml-config";

const config: Config = {
  rules: {
    doctype: true,
  },
};

module.exports = config;
```

### Current behavior
It fetches the specified config file but also contains other presets.
```
2023-04-30T15:08:54.373Z markuplint-cli:ml-engine:file Fetched Config files: Set(9) {
  '/workspaces/markuplint-test/markuplint.config.ts',
  'markuplint:code-styles',
  'markuplint:html-standard',
  'markuplint:a11y',
  'markuplint:performance',
  'markuplint:security',
  'markuplint:rdfa',
  'markuplint:recommended',
  '0'
}
```

### Expected Behavior
It should fetch only one config file, as below.
```
2023-04-30T15:15:41.002Z markuplint-cli:ml-engine:file Fetched Config files: Set(1) { '/workspaces/markuplint-test/markuplint.config.js' }
```

Full test resources and results can be found here.
https://github.com/risu729/markuplint-test/tree/main

**Comment by munierujp** [comment]
We may need to specify `searchPlaces`.

https://github.com/Codex-/cosmiconfig-typescript-loader#usage

**Comment by munierujp** [comment]
@risu729 What about using `export defaukt`?

```ts
import type { Config } from '@markuplint/ml-config'

// https://markuplint.dev/docs/configuration
const config: Config = {
  extends: [
    'markuplint:recommended'
  ]
}

export default config
```

**Comment by risu729** [comment]
@munierujp

I got the same result as using CJS config file.

```
2023-04-30T15:33:16.201Z markuplint-cli:ml-engine:file Fetched Config files: Set(9) {
  '/workspaces/markuplint-test/markuplint.config.ts',
  'markuplint:code-styles',
  'markuplint:html-standard',
  'markuplint:a11y',
  'markuplint:performance',
  'markuplint:security',
  'markuplint:rdfa',
  'markuplint:recommended',
  '0'
}
```

`npx markuplint --config markuplint.config.ts --verbose test.html`

```markuplint.config.ts
import type { Config } from '@markuplint/ml-config'

const config: Config = {
  rules: {
    doctype: true,
  },
};

export default config
```
*I just used `rules` to see whether it's applied easily.

**Comment by munierujp** [comment]
@risu729 Sorry, I've overlooked the following sentence.

>It seems ESM is not supported at all. ([bug: fails to load config in ESM project #60](https://github.com/Codex-/cosmiconfig-typescript-loader/issues/60))

Apparently, it is a different problem than this isssue.

**Comment by risu729** [comment]
@munierujp I agree with you. I would split the issue.

**Comment by YusukeHirao** [comment]
@munierujp @risu729 

Thank you for your reports.

My apologies for leaving the test commented out when I converted it to JS. Could you please improve it to work in several cases, including this one? I appreciate your help, as I couldn't give it enough attention.

https://github.com/markuplint/markuplint/blob/8c7bdb69df903289126fd5e67ee962c193c3714c/packages/%40markuplint/file-resolver/test/config-provider.spec.js#L200-L214

**Comment by YusukeHirao** [comment]
It resolved in #1198
